modules/core/src/main/java/org/apache/synapse/api/API.java

```diff
@@ -477,6 +496,15 @@ void process(MessageContext synCtx) {
         }
     }
 
+    private boolean isBound(Resource resource, MessageContext synCtx) {
```

Comment:
shall we add a small comment on what this method is doing

Reply:
Addressed in 89315e2 and cc02c4c